runtime/oti/jvmtiInternal.h

```diff
@@ -32,19 +32,20 @@
 #define J9_JVMTI_PERFORM_CHECKS
 #endif
 
-#define J9JVMTI_LOWEST_EVENT JVMTI_MIN_EVENT_TYPE_VAL
 #define J9JVMTI_HIGHEST_EVENT JVMTI_MAX_EVENT_TYPE_VAL
 
-typedef enum { 
-	J9JVMTI_BEFORE_FIRST_EXTENSION_EVENT = J9JVMTI_HIGHEST_EVENT,
-	J9JVMTI_EVENT_COM_IBM_COMPILING_START,
-	J9JVMTI_EVENT_COM_IBM_COMPILING_END,
-	J9JVMTI_EVENT_COM_IBM_INSTRUMENTABLE_OBJECT_ALLOC,
-	J9JVMTI_EVENT_COM_IBM_VM_DUMP_START,
-	J9JVMTI_EVENT_COM_IBM_VM_DUMP_END,
-	J9JVMTI_EVENT_COM_IBM_GARBAGE_COLLECTION_CYCLE_START,
-	J9JVMTI_EVENT_COM_IBM_GARBAGE_COLLECTION_CYCLE_FINISH,
-	J9JVMTI_AFTER_LAST_EXTENSION_EVENT
+typedef enum {
+	J9JVMTI_BEFORE_FIRST_EXTENSION_EVENT = JVMTI_MIN_EVENT_TYPE_VAL - 11 ,
+	J9JVMTI_EVENT_COM_IBM_GARBAGE_COLLECTION_CYCLE_START = JVMTI_MIN_EVENT_TYPE_VAL - 10,
+	J9JVMTI_EVENT_COM_IBM_GARBAGE_COLLECTION_CYCLE_FINISH = JVMTI_MIN_EVENT_TYPE_VAL - 9,
+	J9JVMTI_EVENT_COM_IBM_VM_DUMP_START = JVMTI_MIN_EVENT_TYPE_VAL - 8,
+	J9JVMTI_EVENT_COM_IBM_VM_DUMP_END = JVMTI_MIN_EVENT_TYPE_VAL - 7,
+	J9JVMTI_EVENT_COM_IBM_INSTRUMENTABLE_OBJECT_ALLOC = JVMTI_MIN_EVENT_TYPE_VAL - 6,
+	J9JVMTI_EVENT_COM_IBM_COMPILING_START = JVMTI_MIN_EVENT_TYPE_VAL - 5,
+	J9JVMTI_EVENT_COM_IBM_COMPILING_END = JVMTI_MIN_EVENT_TYPE_VAL - 4,
+	J9JVMTI_EVENT_COM_SUN_HOTSPOT_EVENTS_VIRTUAL_THREAD_UNMOUNT = JVMTI_MIN_EVENT_TYPE_VAL - 3,
+	J9JVMTI_EVENT_COM_SUN_HOTSPOT_EVENTS_VIRTUAL_THREAD_MOUNT = JVMTI_MIN_EVENT_TYPE_VAL - 2,
```

Comment:
Why is this -2 and not -1? I dislike this approach - there's no reason to match RI numbering here, since we don't have the same extension events. Also, this limits the number of possible extensions to 50.

Reply:
I suppose the limitation isn't true, but I'd still rather not unnecessarily change the current design.

Reply:
> Why is this -2 and not -1?

`JVMTI_MIN_EVENT_TYPE_VAL - 1` is reserved for the `CLASS_UNLOAD` event. Currently, OpenJ9 defines `CLASS_UNLOAD` as a normal JVMTI event. On 5 August 2003 (0.2.88), `CLASS_UNLOAD` was removed from the JVMTI spec and converted into an extension event. This task is a TODO in OpenJ9 so `JVMTI_MIN_EVENT_TYPE_VAL - 1` is left unused for `CLASS_UNLOAD`.

> unnecessarily change the current design.

The event values need to be the same as the RI since they are passed as input parameters to `jvmtiSetEventNotificationMode`. We will need to meet this requirement for atleast the three extension events, which are supported by the RI: `CLASS_UNLOAD`, `VIRTUAL_THREAD_MOUNT` and `VIRTUAL_THREAD_UNMOUNT`. Otherwise, `jvmtiSetEventNotificationMode` will fail for these extension events.

### Alternate Design
- J9-specific JVMTI extension events can follow the current design.
- RI-specific JVMTI extension events can follow the RI approach.

**Benefit of the alternate design:** For new RI-specific extension events, J9-specific extensions won't need to be re-numbered (shifted left).
**Disadvantage**: Spreading the extension events, before and after the documented/normal JVMTI events, may make some checks complex.

@gacholio Thoughts about the alternate design?

Reply:
https://github.com/eclipse-openj9/openj9/pull/16501#issuecomment-1409211052 contains key points on why we should adopt RI's numbering.